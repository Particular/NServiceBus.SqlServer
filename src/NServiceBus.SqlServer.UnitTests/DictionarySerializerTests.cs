namespace NServiceBus.SqlServer.UnitTests
{
    using System.Collections.Generic;
    using ApprovalTests;
    using NUnit.Framework;
    using Transport.SQLServer;

    [TestFixture]
    public class DictionarySerializerTests
    {

        [Test]
        public void Can_round_trip()
        {
            var before = new Dictionary<string, string>
            {
                {"key1", "value1"},
                {"key2", "value2"},
                {"keyWithNull", null},
                {"keyWithEmpty", ""}
            };
            var serialize = DictionarySerializer.Serialize(before);
            Approvals.Verify(serialize);

            var after = DictionarySerializer.DeSerialize(serialize);

            AssertDictionariesAreTheSame(before, after);
        }

        void AssertDictionariesAreTheSame(Dictionary<string, string> before, Dictionary<string, string> after)
        {
            foreach (var beforeitem in before)
            {
                Assert.AreEqual(beforeitem.Value, after[beforeitem.Key]);
            }
            Assert.AreEqual(before.Count, after.Count);
        }
    }
}