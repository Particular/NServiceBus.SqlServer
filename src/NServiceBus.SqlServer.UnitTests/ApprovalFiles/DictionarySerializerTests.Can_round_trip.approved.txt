{"key1":"value1","key2":"value2","keyWithNull":null,"keyWithEmpty":""}